lib/utils.pm

```diff
@@ -841,6 +841,10 @@ sub random_string {
 }
 
 sub handle_login {
```

Comment:
Please put a POD doc above in format as e.g. for `assert_screen_with_soft_timeout` and describe what the variables are for. I do not fully understand the purpose of `$user_selected`